akka-slf4j/src/main/scala/akka/event/slf4j/AkkaLogbackClassicAdaptingFilter.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.event.slf4j
+
+import ch.qos.logback.classic.spi.LoggingEvent
+import ch.qos.logback.core.filter.Filter
+import ch.qos.logback.core.spi.FilterReply
+
+/**
+ * In Akka typed actors the log event timestamp and thread is correct since logging is done directly using SLF4J, much of the
+ * Akka internals however use the classic Akka logging api which is asynchronous and passes over the message bus, and for that
+ * source thread and timestamp has special handling, ending up in two MDC entries.
+ *
+ * This filter adapts classic log events by picking up the `sourceThread` and `akkaTimestampMillis` MDC values
+ * and puts those in the respective fields of the log event.
+ */
+class AkkaLogbackClassicAdaptingFilter extends Filter[LoggingEvent] {
+
+  def decide(event: LoggingEvent): FilterReply = {
+    val mdc = event.getMDCPropertyMap
+
+    // FIXME what about thread safety modifying the events? needs to be synchronized?
+    if (mdc.containsKey("sourceThread")) {
+      // classic emitted event, try to convert to
+      event.setThreadName(mdc.get("sourceThread"))
```

Comment:
I don't think this is what filters are supposed to do.
https://jira.qos.ch/browse/LOGBACK-596

Even if it works I'm not sure we want to provide such hack.

A new layout/encoder can probably do this (by extending `PatternLayout`), but that's only solving it for output to files and such. Logstash would need something else.

For the "sourceThread" it would be easiest to just include that in the MDC also from Typed LoggingContext.

For the timestamp I think we can just ignore the problem and say that ordinary timestamp is good enough.

Reply:
There isn't really any intended API in logback for transforming the log events, so yes, indeed a hack. Very questionable.

That linked issue not really related though because it is about `TurboFilter` - the filters invoked to help avoid creating the event.

I think putting it in a duplicate MDC field for typed actors is fine, let's go that path instead.